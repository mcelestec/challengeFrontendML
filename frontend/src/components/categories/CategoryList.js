import React from 'react'
import PropTypes from 'prop-types';

const CategoryList = ({ items }) => {
    return (
        <div className="category-list__container">
            {
                items.map(category =>
                    <span className="category-list__item" key={category} data-testid="category-list-item">{category}</span>
                )
            }
        </div>
    )
}

CategoryList.propTypes = {
    items: PropTypes.arrayOf(PropTypes.string).isRequired
}

export default CategoryList