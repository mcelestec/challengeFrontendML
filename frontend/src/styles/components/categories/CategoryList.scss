.category-list__container {
    margin: 1rem 0;
    min-width: 1048px;

    @include lg {
        padding: 0 10px;
        min-width: 0;
    }
}

.category-list__item {
    margin-right: 5px;
    font-size: 0.875rem;
    color: $middle-dark-gray;

    &::after {
        content: '>';
        margin-left: 5px;
    }

    &:last-child {
        font-weight: 700;

        &::after {
            content: '';
        }
    }
}