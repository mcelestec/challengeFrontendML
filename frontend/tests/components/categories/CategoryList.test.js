import { render, screen } from "@testing-library/react";
import CategoryList from "../../../src/components/categories/CategoryList";

describe('Test: probando <CategoryList />', () => {
    const items = ['Category 1', 'Category 2', 'Category 3', 'Category 4'];

    test('Llamada a CategoryList: debe hacer match con el snapshot', () => {
        const { container } = render(<CategoryList items={items} />);
        expect(container).toMatchSnapshot();
    }
    );

    test('Llamada a CategoryList: probar que se muestren todos los elementos de la lista', () => {

        render(<CategoryList items={items} />);

        const categoryElements = screen.getAllByTestId('category-list-item');
        expect(categoryElements.length).toBe(4);

        // validar si estÃ¡n todas las categorias
        const categories = [];
        categoryElements.forEach(i => categories.push(i.innerHTML));
        expect(items.every(i => categories.includes(i))).toBeTruthy();
    }
    );

});