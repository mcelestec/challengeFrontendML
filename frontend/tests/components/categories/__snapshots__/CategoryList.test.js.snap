// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: probando <CategoryList /> Llamada a CategoryList: debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="category-list__container"
  >
    <span
      class="category-list__item"
      data-testid="category-list-item"
    >
      Category 1
    </span>
    <span
      class="category-list__item"
      data-testid="category-list-item"
    >
      Category 2
    </span>
    <span
      class="category-list__item"
      data-testid="category-list-item"
    >
      Category 3
    </span>
    <span
      class="category-list__item"
      data-testid="category-list-item"
    >
      Category 4
    </span>
  </div>
</div>
`;
